library/Jira/RestApi.php

```diff
@@ -123,9 +123,32 @@ public function updateIssue(IssueUpdate $update)
         return $this->put('issue/' . urlencode($update->getKey()), $update->toObject());
     }
 
+    protected function prepareProjectIssueQuery($project, $host = null, $service = null, $onlyOpen = true)
+    {
+        $query = "creator = currentUser() AND project = $project";
+
+        if ($onlyOpen) {
+            $query .= ' AND resolution is empty';
+        }
+
+        if ($host === null) {
+            $query .= ' AND icingaKey ~ "BEGIN*"';
+        } else {
+            $icingaKey = static::makeIcingaKey($host, $service);
+
+            // There is no exact field matcher out of the box on JIRA, this is
+            // an ugly work-around. We search for "BEGINhostnameEND" or
+            // "BEGINhostname!serviceEND"
+            $query .= \sprintf(' AND icingaKey ~ "\"%s\""', $icingaKey);
+        }
+
+        $query .= ' ORDER BY created DESC';
```

Comment:
Please wrap this in a private method, e.g. `finalizeQuery($host, $service)` and call it in both methods.